core/settings.lua

```diff
@@ -68,6 +68,13 @@ function settings:_init()
     help = "Skip to be added to left side of line"
   })
 
+  self:declare({
+    parameter = "document.zenkakuchar",
```

Comment:
Setting scopes are an open issue (#1327), but should this be a document level setting or might it be better off scoped to the typesetter or ... something else? I'll probably go ahead with this merge this way since rearranging that mess is going to be a breaking change anyway, but just looking for your idea on what this *should* be.

Reply:
We should expect the user to swap it mid-document. I guess that means scoped to the typesetter.

I'm a Unicode nerd so I tend to think of the rarer languages of the world, and I can imagine documents where you'd want to flip to Ainu or Taiwanese Kana or Okinawan. All three of those are written in Japanese kana and with traditional Japanese typesetting rules including _zenkaku_. But we shouldn't expect all fonts made _just_ for e.g. Ainu or Okinawan to support even あ, as the Ainu orthography (example: アイヌ・イタㇰ) doesn't include hiragana.

Reply:
Or scoped to the font? (perhaps it could even be something extracted from font switching commands, rather than via settings?)

Reply:
I know nothing about eastern languages, but isn't U+3000 IDEOGRAPHIC SPACE supposed to provide that width?

Reply:
Scoping to the font won't work. The OT feature `ruby` often has a different zenkaku width. It's typesetter scope if anything.

U+3000 would be more workable than あ and is an interesting suggestion of a better default.

Reply:
Scoping to the font would only make sense if the unit was something only used to scale the font itself. In this case it's a general purpose unit that get used for all sorts of things relative to much larger scopes including frame properties. Typesetter makes a lot of sense I think. Although we do have a bunch of document level properties that are actually at the same scope that are not immutable class options, so `document._` for now until we sort that out sounds like a fair plan.